nodestream_plugin_neptune/ingest_query_builder.py

```diff
@@ -0,0 +1,245 @@
+    # There must be better ways to handle these.
+    # It's here only for the PoC.
+    for k, v in props.items():
+        # Neptune can handles datetime type
+        # see: https://docs.aws.amazon.com/neptune/latest/userguide/feature-opencypher-compliance.html#opencypher-compliance-differences
+        if isinstance(v, Timestamp):
+            props[k] = str(v)
+        # Could the nodestream filter/transforms these values else where?
+        elif not v:
+            props[k] = 'None'
+        elif isinstance(v, numbers.Number) and math.isnan(v):
+            props[k] = "NaN"
+
+    return props
+
+
+class NeptuneDBIngestQueryBuilder:
+    @cache
+    @correct_parameters
+    def generate_update_node_operation_query_statement(
+        self,
+        operation: OperationOnNodeIdentity,
+        ref: str,
+    ) -> str:
+        """Generate a query to update a node in the database given a node type and a match strategy."""
+        labels = ":".join([
+            operation.node_identity.type,
+            *operation.node_identity.additional_types,
+        ])
+        node_id_param_name = generate_id_param_name(GENERIC_NODE_REF_NAME)
+        merge_node = _merge_node(labels, node_id_param_name, GENERIC_NODE_REF_NAME)
+
+        """
+        At this time, Neptune doesn't support nested maps very well.
+        We get an error trying to access inner maps in the parameters
+        of our openCypher query. 
+        
+        As such, node's properties and __node_id has to be kept at 
+        the same level.
+
+        As a work around, we use removeKeyFromMap() to remove __node_id before 
+        setting node's properties. removeKeyFromMap() Neptune specific.
+        """
+        on_create = f"""ON CREATE SET {GENERIC_NODE_REF_NAME} = removeKeyFromMap(param, "{node_id_param_name}")"""
+        on_match = f"""ON MATCH SET {GENERIC_NODE_REF_NAME} += removeKeyFromMap(param, "{node_id_param_name}")"""
+        query = f"{merge_node} {on_create} {on_match}"
+        return query
+
+    def generate_update_node_operation_params(self, node: Node) -> dict:
+        """Generate the parameters for a query to update a node in the database."""
+
+        node_props = {**self.generate_node_key_params(node), **node.properties}
+        node_props = _to_string_values(node_props)
+
+        return node_props
+
+    def generate_node_key_params(self, node: Node, name=GENERIC_NODE_REF_NAME) -> dict:
+        """Generate the parameters for a query to update a node in the database."""
+
+        # Todo: What if no keys were given? Maybe let neptune decides.
+        # On uniqueness and keys in Neptune, see Schema Constraints in https://docs.aws.amazon.com/neptune/latest/userguide/migration-compatibility.html
+        composite_key = "_".join([str(node.key_values[k]) for k in node.key_values])
+        composite_key = f"{node.type}_{composite_key}"
+        return {generate_prefixed_param_name("id", name): composite_key}
```

Comment:
My original comment was going to be `nodestream` requires that `Node` passed in has a defined and non-null key which would sidestep this issue. However, its actually only validating that the key is non-null. See [here](https://github.com/nodestream-proj/nodestream/blob/4ceb4e8fee8fd9dd1f3f8c8d0b3eef8286f41845/nodestream/model/graph_objects.py#L91).  I have opened https://github.com/nodestream-proj/nodestream/issues/248 to address this. 

I'd recommend working under the assumption this is not a case that needs to be dealt with and we'll address it in nodestream core.